### PR TITLE
Enable GitHub teams for the mod team

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @pietroalbini

### CODE REVIEW

**Review by ashleygwilliams — APPROVED** [review]

File: teams/mods.toml [hunk 1]
```diff
@@ -9,6 +9,9 @@ members = [
     "matthieu-m",
 ]
 
+[[github]]
```
Comment (pietroalbini) [inline teams/mods.toml:12]: On GitHub the team is currently called `moderation`, so we either add `team-name = "moderation"` or we rename the existing team.

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: teams/mods.toml (hunk 1 above)
Comment (Mark-Simulacrum) [reply teams/mods.toml:12]: I'd rather rename the github team, I'd prefer to have everything consistent with the names used in the team repo. Ultimately having two teams is also not the end of the world.

Comment (pietroalbini) [reply teams/mods.toml:12]: Ok, renamed the team in the GitHub UI! Resolving this comment.

**Review by pietroalbini — APPROVED** [review]

**Comment by camelid** [comment]
I think the rfcbot ping (and `[website]` team name) are now incorrect because of the GitHub team rename:

```
[rfcbot]
label = "T-moderation"
name = "Moderation"
ping = "rust-lang/moderation"

[website]
page = "moderation"
```

**Pull request merged**